strax/storage/s3.py

```diff
@@ -88,26 +92,26 @@ def _find(self, key, write, fuzzy_for, fuzzy_for_options):
         if fuzzy_for or fuzzy_for_options:
             raise NotImplementedError("Can't do fuzzy with S3")
 
-        # Check exact match / write case
         key_str = str(key)
         bk = self.backend_key(key_str)
 
-        # See if any objects exist for this key
-        objects_list = self.s3.list_objects(Bucket=BUCKET_NAME,
-                                            Prefix=key_str)
-        if 'Contents' in objects_list:
-            if write and not self._can_overwrite(key):
-                raise strax.DataExistsError(at=bk)
-            return bk
-        else:
-            # No objects yet...
-            if write:
-                return bk
+        try:
+            self.backends[0].get_metadata(key)
+        except ClientError as ex:
+            if ex.response['Error']['Code'] == 'NoSuchKey':
```

Comment:
Could you catch this in get_metadata and raise DataNotAvailable there? This would fix #90. (Then here, you'd catch DataNotAvailable again if you're writing.)

The S3 frontend is unusual in that `_find` relies on `get_metadata` rather than the other way around. But the context relies on DataNotAvailable being raised if get_metadata fails, so it can try other storage frontends (https://github.com/AxFoundation/strax/blob/master/strax/context.py#L632).

Reply:
Actually it turns out #90 has already been fixed, so never mind.